app/src/main/java/com/clint/nasa/core/exception/Failure.kt

```diff
@@ -0,0 +1,6 @@
+package com.clint.nasa.core.exception
+
+sealed class Failure {
+    object NetworkConnection : Failure()
+    object ServerError : Failure()
```

Comment:
What will happen if we need a feature specific error handling.?

Reply:
We can have something like an abstract class here. Then, anyone can extend that abstract class for feature specific error messages. Will add this if the need arises
/** * Extend this class for feature specific failures.*/
    `abstract class FeatureFailure : Failure()`